index.js

```diff
@@ -4,49 +4,113 @@ const app = express()
+
+app.use(express.json({extended: true, limit: '1mb'}))
 
 app.all('/', (req, res) => {
     console.log("Just got a request!")
     res.send('Yo!')
 })
 
-let file_context = ""
-fs.readFile("./file_context.txt", 'utf8', function(err, data) {
-  if (err) throw err;
-  console.log(file_context);
-  file_context = data;
-});
+if (process.env.GPT_MODE === "CHAT"){
+
+  fs.readFile("./file_context.txt", 'utf8', function(err, data) {
+    if (err) throw err;
+    console.log("Reading context file and adding it as system level message for the agent.")
+    messages[0].content = data;
+  });
+
+} else {
+
+  fs.readFile("./file_context.txt", 'utf8', function(err, data) {
+    if (err) throw err;
+    console.log("Reading context file and adding it in front of user prompts:")
+    file_context = data;
+    console.log(file_context);
+  });
+
+}
 
 app.get('/gpt/:text', async (req, res) => {
+    
+    //The agent should recieve Username:Message in the text to identify conversations with different users in his history. 
+    
     const text = req.params.text
     const { Configuration, OpenAIApi } = require("openai");
 
-    console.log(process.env.OPENAI_API_KEY)
     const configuration = new Configuration({
       apiKey: process.env.OPENAI_API_KEY,
     });
-    const openai = new OpenAIApi(configuration);
 
-    const prompt = file_context + "\n\nQ:" + text + "\nA:";
-    console.log(prompt);
+    const openai = new OpenAIApi(configuration);      
     
-    const response = await openai.createCompletion({
-      model: "text-davinci-003",
-      prompt: prompt,
-      temperature: 0.5,
-      max_tokens: 128,
-      top_p: 1,
-      frequency_penalty: 0,
-      presence_penalty: 0,
-    });
-    if (response.data.choices) {
-        res.send(response.data.choices[0].text)
-    } else {
+    if (GPT_MODE === "CHAT"){
+      //CHAT MODE EXECUTION
+
+      //Add user message to  messages
+      messages.push({role: "user", content: text})
+      //Check if message history is exceeded
+      console.log("Conversations in History: " + ((messages.length / 2) -1) + "/" + process.env.HISTORY_LENGTH)
+      if(messages.length > ((process.env.HISTORY_LENGTH * 2) + 1)) {
+          console.log('Message amount in history exceeded. Removing oldest user and agent messages.')
+          messages.splice(1,2)
+     }
+    
+      console.log("Messages: ")
+      console.dir(messages)
+      console.log("User Input: " + text)
+
+      const response = await openai.createChatCompletion({
+        model: "gpt-3.5-turbo",
+        messages: messages,
+        temperature: 0.5,
+        max_tokens: 128,
+        top_p: 1,
+        frequency_penalty: 0,
+        presence_penalty: 0,
+      });
+    
+      if (response.data.choices) {
+        console.log ("Agent answer: " + response.data.choices[0].message.content)
+        messages.push({role: "assistant", content: response.data.choices[0].message.content})
+        res.send(response.data.choices[0].message.content)
```

Comment:
Could you also add in this PR a .slice(0,399) to not exceed the twitch 400 characters limit?

Thank you

Reply:
Actually we could slice the response up into parts of 400 characters and send multiple times if a response goes over that limit. I will add it.